DSP Accel Library/Testing and Images/TestingLibraryScripts.m

```diff
@@ -0,0 +1,129 @@
+
+    %%%%%%%%%% check and identify error %%%%%%%%%%%
+    % Make some values for an "ideal" lookup table with log spaced points
+    X_in_Ideal = logspace(log10(min_val), log10(max_val), 2^(ram_size+2));
+    X_temp = X_in;
+    X_in = X_in_Ideal;
+    % identify the values of the function at X_in_Ideal
+    Y_out_Ideal = eval(TableFn);
+
+    % Get values for lookup table (already have X_in)
+    X_in = X_temp;
+
+    % Find lookup addresses for each point in X_in_Ideal
+    x_addr = zeros(1,length(X_in_Ideal));
+    for it = 1:length(X_in_Ideal)
+        x_addr(it) = 2^ram_size;
+        X_Shift = X_in_Ideal(it);
+        while(X_Shift < X_in(x_addr(it)) && x_addr(it) ~= 1)
+            x_addr(it) = x_addr(it) - 1;
+        end
+        if(x_addr(it)==0)
+            x_addr(it) = 1;
+        end
+    end
+
+    % Check for any possible out of bounds errors (handled similarly in
+    % hardware)
+    x_addr(x_addr == 2^ram_size) = 2^ram_size -1;
+
+    % Get values for linear interpolation of X_in_Ideal
+    x_low  = X_in(x_addr);
+    x_high = X_in(x_addr+1);
+
+
+    y_low  = Y_out(x_addr);
+    y_high = Y_out(x_addr+1);
+    slope  = (y_high-y_low) ./ (x_high - x_low);
+    y_inter = slope.*(X_in_Ideal-x_low)+y_low;
+    % percent error: obt-exp / exp
+    %errorFloor = (Y_out_Ideal-y_low)./Y_out_Ideal;
+    errorInter = (Y_out_Ideal-y_inter)./Y_out_Ideal;
+    %maxFloorErr(ix) = 100*max(abs(errorFloor(ix, 2870:9839)));
+    %maxInterErr(ix) = 100*max(abs(errorInter(ix, 2870:9839)));
+    maxErr = max(abs(errorInter));
+    
+    %check while loop condition
+    if(igotthis) 
+        repeatFlag = false;
+    else
+        if(maxErr <= error_cap_tab)
+            repeatFlag = false;
+            if(ERR_DIAG)
+                figure(1); 
+                subplot(2,1,1); semilogx(X_in_Ideal,y_inter, X_in_Ideal,Y_out_Ideal,X_in,Y_out,'k*'); title('Output Values over Input Range'); xlabel('Inputs'); ylabel('Outputs'); legend('Interpolated','Ideal','Table Points');
+                subplot(2,1,2); semilogx(X_in_Ideal,100*errorInter); title('Error of Output over Input Range'); xlabel('Inputs'); ylabel('Percent Error');
+            end
+        else
+            if(ERR_DIAG)
+                f = msgbox(sprintf('Err = %.2d, M_bits = %d', maxErr, M_bits),'Configuring PLUT','replace');
+            end
+            if(maxErr > 8*error_cap_tab)
+                M_bits = M_bits+3;
+            elseif(maxErr > 4*error_cap_tab)
+                M_bits = M_bits+2;
+            else
+                M_bits = M_bits+1;
+            end
+        end
+    end
```

Comment:
A few comments here would be good as well.  Is there a case where igotthis = true, but you still want to generate some plots?